themes/default/content/blog/converting-bicep-to-pulumi/index.md

```diff
@@ -0,0 +1,357 @@
+
+## Building your own Pulumi language converter
+
+Converter plugins are a great way to extend the Pulumi CLI with `pulumi convert` to support new languages. When we shipped support for [Converting full Terraform Programs To Pulumi](https://www.pulumi.com/blog/converting-full-terraform-programs-to-pulumi/), we extended the `pulumi convert` command to allow installing and using converter plugins that are shipped independently from the Pulumi CLI.
+
+In essense, those plugins are executable binaries that can be written in any language which serves up a gRPC server that implements the `Converter` contract:
+
+```protobuf
+// Converter is a service for converting between other ecosystems and Pulumi.
+// This is currently unstable and experimental.
+service Converter {
+    // ConvertState converts state from the target ecosystem into a form that can be imported into Pulumi.
+    rpc ConvertState(ConvertStateRequest) returns (ConvertStateResponse) {}
+
+    // ConvertProgram converts a program from the target ecosystem into a form that can be used with Pulumi.
+    rpc ConvertProgram(ConvertProgramRequest) returns (ConvertProgramResponse) {}
+}
+```
+
+> This Converter interface is considered experimental and is subject to change in the future.
+
+Now, learning how to setup gRPC servers and implement the `Converter` contract is not an easy task unless you are familiar with it. For dotnet, we shipped the _experimental_ Pulumi converter SDK to the main Pulumi nuget package that makes it extremely easy to build converter plugins.
+
+A bare-bones converter plugin which does nothing looks this in F#:
+
+```fsharp
+open Pulumi.Experimental.Converter
+open Pulumi.Codegen
+
+let convertProgram (request: ConvertProgramRequest): ConvertProgramResponse =
+    // TODO: Implement the conversion logic here
+    ConvertProgramResponse.Empty
+
+convertProgram
+|> Converter.CreateSimple
+|> Converter.Serve
+|> Async.AwaitTask
+|> Async.RunSynchronously
+```
+
+This will spin up a gRPC server that implements the `Converter` contract and will serve it on a random port assigned during startup. The Pulumi converter SDK automatically implements gRPC reflection which allows tools such as [Postman](https://www.postman.com/) to discover the available gRPC services and methods and send example requests to them. See [Postman with gRPC](https://blog.postman.com/postman-now-supports-grpc/).
+
+## Publishing converter plugins
+
+The easiest way to make your converter plugin available to others is to publish it on GitHub releases. The Pulumi CLI expects a naming convention `pulumi-converter-<language>` and will automatically download the plugin from the latest GitHub release of the repository when given the `--server` flag during `pulumi plugin install converter <language> --server <url>`.
```

Comment:
Maybe use \<name\> instead of \<language\>? Bit confusing that maybe this is talking about typescript/csharp/etc

Reply:
Yeah `<name>` is better, adjusted ✅